proof/crefine/ARM/Schedule_C.thy

```diff
@@ -357,11 +357,20 @@ lemma nextDomain_ccorres:
   apply simp
   done
 
+lemma prepareNextDomain_ccorres:
+  "ccorres dc xfdc invs' UNIV [] prepareNextDomain (Call prepareNextDomain_'proc)"
+  apply cinit
+  apply (rule ccorres_return_Skip)
+  by clarsimp
+
 lemma scheduleChooseNewThread_ccorres:
   "ccorres dc xfdc
      (\<lambda>s. invs' s \<and> ksSchedulerAction s = ChooseNewThread) UNIV hs
      (do domainTime \<leftarrow> getDomainTime;
-         y \<leftarrow> when (domainTime = 0) nextDomain;
+         y \<leftarrow> when (domainTime = 0) (do
+             y <- prepareNextDomain;
+             nextDomain
```

Comment:
indent question: shouldn't the terms under the `when` be indented more than the `when`, hence also the following `od`?

Reply:
Hmm, this is a confusing case with the variable binding but I think it's right. If we ignore the spurious variables then the `od` is the same indentation as the `when`, and the terms inside it are indented further like you say. This also matches what we were happy with in the abstract:

https://github.com/seL4/l4v/blob/eeafbc3e6fa018186c38e2544a85fc00a0ddb480/spec/abstract/Schedule_A.thy#L94-L97